### PR TITLE
Building Road Intersection Enhancements

### PR DESCRIPTION
**Pull request by savannahostrowski**
The Building Road Intersection (BRI) integrity check is meant to analyze overlap between roads and buildings. Barring exceptions for certain scenarios, the geometry for a road and that for a building’s boundary are not supposed to intersect each other.

This PR depends on Atlas PR [#110](https://github.com/osmlab/atlas/pull/110) for the AmenityTag.PARKING_ENTRANCE enum, and Atlas PR [#121](https://github.com/osmlab/atlas/pull/121) for the EntranceTag.java functionality.

**Added functionality:**
- Ignore tag amenity=parking
- Ignore roads/buildings who intersect at a Node which is labeled Parking Garage Entrance/Exit or entrance=yes
- Ignore tag area=yes
- Ignore tag indoor=yes
- Ignore tag tunnel=yes
- Ignore tag highway=service when intersecting building with tag amenity=fuel
- Ignore Ways with tags highway=service and service=driveway

**OSM Examples:**
Example 1:
- Building Feature: https://www.openstreetmap.org/way/310947685
- Road Feature: https://www.openstreetmap.org/way/189212967

Example 2:
- Building Feature: https://www.openstreetmap.org/way/108232856
- Road Feature: https://www.openstreetmap.org/way/525007924

### CODE REVIEW

File: src/main/java/org/openstreetmap/atlas/checks/validation/intersections/BuildingRoadIntersectionCheck.java [hunk 1]
```diff
@@ -33,25 +36,44 @@
...
     private static final long serialVersionUID = 5986017212661374165L;
 
     private static Predicate<Edge> ignoreTags()
     {
         return edge -> !(Validators.isOfType(edge, CoveredTag.class, CoveredTag.YES)
-                || Validators.isOfType(edge, TunnelTag.class, TunnelTag.BUILDING_PASSAGE));
+                || Validators.isOfType(edge, TunnelTag.class, TunnelTag.BUILDING_PASSAGE,
+                        TunnelTag.YES)
+                || Validators.isOfType(edge, AreaTag.class, AreaTag.YES)
+                || (edge.getTag(INDOOR_KEY).isPresent() && edge.tag(INDOOR_KEY).equals(YES_VALUE))
```
Comment (mertemin) [inline src/main/java/org/openstreetmap/atlas/checks/validation/intersections/BuildingRoadIntersectionCheck.java]: This condition could be improved with `YES_VALUE.equals(edge.tag(INDOOR_KEY))`. Non-null expected value can be used as reference and that way null values could be handled without extra presence check.

File: src/main/java/org/openstreetmap/atlas/checks/validation/intersections/BuildingRoadIntersectionCheck.java [hunk 2]
```diff
@@ -70,12 +92,7 @@ private static boolean isValidIntersection(final Area building, final Edge edge)
                 && !building.asPolygon().fullyGeometricallyEncloses(edge.asPolyLine()))
         {
             if (intersections.contains(edgeStart.getLocation())
-                    && edge.inEdges().stream().anyMatch(ignoreTags().negate()))
-            {
-                return true;
-            }
-            if (intersections.contains(edgeEnd.getLocation())
-                    && edge.outEdges().stream().anyMatch(ignoreTags().negate()))
+                    || intersections.contains(edgeEnd.getLocation()))
```
Comment (mertemin) [inline src/main/java/org/openstreetmap/atlas/checks/validation/intersections/BuildingRoadIntersectionCheck.java]: This if block could be combined with the outer if block.

**Comment by mgcuthbert** [comment]
We will hold off merging this until we have the dependencies from Atlas squared away.

**Comment by danielduhh** [comment]
This should be good now - https://github.com/osmlab/atlas/blob/dev/src/main/java/org/openstreetmap/atlas/tags/AmenityTag.java#L15

**Pull request merged**